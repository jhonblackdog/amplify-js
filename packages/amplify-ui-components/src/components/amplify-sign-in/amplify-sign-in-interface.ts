export interface AmplifyForgotPasswordHintProps {
  forgotPasswordText: string;
  resetPasswordText: string;
}

export interface SignInFormFooterProps {
  submitButtonText: string;
  noAccountText: string;
  createAccountText: string;
  overrideStyle?: boolean;
}