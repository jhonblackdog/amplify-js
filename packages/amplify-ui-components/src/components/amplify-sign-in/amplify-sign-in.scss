:host {
  --footer-size: var(--amplify-text-sm);
  --footer-color: var(--amplify-grey);
  --footer-font-family: var(--amplify-font-family);
}

.sign-in-form-footer {
  font-family: var(--footer-font-family);
  font-weight: 300;
  font-size: var(--footer-size);
  color: var(--footer-color);
  margin-bottom: 10px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: baseline;
}
