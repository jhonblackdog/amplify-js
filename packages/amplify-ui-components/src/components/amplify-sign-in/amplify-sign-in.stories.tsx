import { storiesOf } from '@storybook/html';
import { knobs } from '../../common/testing';

const selectStories = storiesOf('amplify-sign-in', module);

selectStories.add('default', () => {
  const override = knobs.overrideStyleKnob();
  return `<amplify-sign-in override-style=${override}></amplify-sign-in>`;
});
