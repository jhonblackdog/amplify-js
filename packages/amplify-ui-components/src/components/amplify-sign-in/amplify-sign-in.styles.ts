import { css } from 'emotion';

export const signInFormFooter = css`
  font-family: var(--font-family);
  font-weight: 300;
  font-size: var(--secondary-font-size);
  color: var(--grey);
  margin-bottom: 10px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: baseline;
`;
