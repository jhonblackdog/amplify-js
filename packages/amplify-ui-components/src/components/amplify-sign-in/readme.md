# amplify-sign-in



<!-- Auto Generated Below -->


## Properties

| Property           | Attribute            | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | Type                         | Default              |
| ------------------ | -------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------- | -------------------- |
| `formFields`       | --                   | Form fields allows you to utilize our pre-built components such as username field, code field, password field, email field, etc. by passing an array of strings that you would like the order of the form to be in. If you need more customization, such as changing text for a label or adjust a placeholder, you can follow the structure below in order to do just that. ``` [   {     type: 'username'\|'password'\|'email'\|'code'\|'default',     label: string,     placeholder: string,     hint: string \| Functional Component \| null,     required: boolean   } ] ``` | `FormFieldTypes \| string[]` | `SIGN_IN_COMPONENTS` |
| `handleSubmit`     | --                   | Fires when sign in form is submitted                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | `(Event: any) => void`       | `undefined`          |
| `headerText`       | `header-text`        | Used for header text in sign in component                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | `string`                     | `HEADER_TEXT`        |
| `overrideStyle`    | `override-style`     | (Optional) Overrides default styling                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | `boolean`                    | `false`              |
| `submitButtonText` | `submit-button-text` | Used for the submit button text in sign in component                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | `string`                     | `SUBMIT_BUTTON_TEXT` |
| `validationErrors` | `validation-errors`  | Engages when invalid actions occur, such as missing field, etc.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | `string`                     | `undefined`          |


## Dependencies

### Used by

 - [amplify-authenticator](../amplify-authenticator)
 - [amplify-examples](../amplify-examples)

### Depends on

- [amplify-form-section](../amplify-form-section)
- [amplify-auth-fields](../amplify-auth-fields)
- [amplify-link](../amplify-link)
- [amplify-link](../amplify-link)
- [amplify-button](../amplify-button)

### Graph
```mermaid
graph TD;
  amplify-sign-in --> amplify-form-section
  amplify-sign-in --> amplify-auth-fields
  amplify-sign-in --> amplify-link
  amplify-sign-in --> amplify-link
  amplify-sign-in --> amplify-button
  amplify-form-section --> amplify-section
  amplify-form-section --> amplify-button
  amplify-auth-fields --> amplify-username-field
  amplify-auth-fields --> amplify-password-field
  amplify-auth-fields --> amplify-email-field
  amplify-auth-fields --> amplify-code-field
  amplify-auth-fields --> amplify-form-field
  amplify-auth-fields --> amplify-form-field
  amplify-username-field --> amplify-form-field
  amplify-form-field --> amplify-label
  amplify-form-field --> amplify-input
  amplify-form-field --> amplify-hint
  amplify-password-field --> amplify-form-field
  amplify-email-field --> amplify-form-field
  amplify-code-field --> amplify-form-field
  amplify-authenticator --> amplify-sign-in
  amplify-examples --> amplify-sign-in
  style amplify-sign-in fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
