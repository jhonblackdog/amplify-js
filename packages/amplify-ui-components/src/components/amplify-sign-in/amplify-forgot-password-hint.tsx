import { FunctionalComponent as FC, h } from '@stencil/core';
import { AmplifyForgotPasswordHintProps } from './amplify-sign-in-interface';

export const AmplifyForgotPasswordHint: FC<AmplifyForgotPasswordHintProps> = ({ forgotPasswordText, resetPasswordText }) => (
  <div>
    {forgotPasswordText} <amplify-link>{resetPasswordText}</amplify-link>
  </div>
);
