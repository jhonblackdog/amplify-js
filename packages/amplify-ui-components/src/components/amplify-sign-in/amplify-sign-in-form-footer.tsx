import { FunctionalComponent as FC, h } from '@stencil/core';
import { AmplifySignInFormFooterProps } from './amplify-sign-in-interface';
import { signInFormFooter } from './amplify-sign-in.styles';

export const AmplifySignInFormFooter: FC<AmplifySignInFormFooterProps> = ({ submitButtonText, noAccountText, createAccountText, overrideStyle = false }) => (
  <div class={signInFormFooter}>
    <span>{noAccountText} <amplify-link>{createAccountText}</amplify-link></span>
    <amplify-button type="submit" overrideStyle={overrideStyle}>{submitButtonText}</amplify-button>
  </div>
);
