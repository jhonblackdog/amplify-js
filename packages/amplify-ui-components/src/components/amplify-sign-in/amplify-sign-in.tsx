import { Component, FunctionalComponent as FC, Prop, h } from '@stencil/core';
import { AmplifyForgotPasswordHintProps, SignInFormFooterProps } from './amplify-sign-in-interface';
import { FormFieldTypes } from '../../components/amplify-auth-fields/amplify-auth-fields-interface';
import { signInFormFooter } from './amplify-sign-in.styles';
import {
  HEADER_TEXT,
  SUBMIT_BUTTON_TEXT,
  CREATE_ACCOUNT_TEXT,
  NO_ACCOUNT_TEXT,
  FORGOT_PASSWORD_TEXT,
  RESET_PASSWORD_TEXT
} from '../../common/constants';


export const AmplifyForgotPasswordHint: FC<AmplifyForgotPasswordHintProps> = ({ forgotPasswordText, resetPasswordText }) => (
  <div>
    {forgotPasswordText} <amplify-link>{resetPasswordText}</amplify-link>
  </div>
);

const SignInFormFooter: FC<SignInFormFooterProps> = ({ submitButtonText, noAccountText, createAccountText, overrideStyle = false }) => (
  <div class={signInFormFooter}>
    <span>{noAccountText} <amplify-link>{createAccountText}</amplify-link></span>
    <amplify-button type="submit" overrideStyle={overrideStyle}>{submitButtonText}</amplify-button>
  </div>
);

const SIGN_IN_COMPONENTS = [
  {
    type: 'username',
    required: true,
  },
  {
    type: 'password',
    hint: <AmplifyForgotPasswordHint forgotPasswordText={FORGOT_PASSWORD_TEXT} resetPasswordText={RESET_PASSWORD_TEXT} />,
    required: true,
  },
];

@Component({
  tag: 'amplify-sign-in',
  shadow: false,
})
export class AmplifySignIn {
  /** Fires when sign in form is submitted */
  @Prop() handleSubmit: (Event) => void;
  /** Engages when invalid actions occur, such as missing field, etc. */
  @Prop() validationErrors: string;
  /** Used for header text in sign in component */
  @Prop() headerText: string = HEADER_TEXT;
  /** Used for the submit button text in sign in component */
  @Prop() submitButtonText: string = SUBMIT_BUTTON_TEXT;
  /** (Optional) Overrides default styling */
  @Prop() overrideStyle: boolean = false;
  /** 
   * Form fields allows you to utilize our pre-built components such as username field, code field, password field, email field, etc.
   * by passing an array of strings that you would like the order of the form to be in. If you need more customization, such as changing
   * text for a label or adjust a placeholder, you can follow the structure below in order to do just that.
   * ```
   * [
   *  {
   *    type: 'username'|'password'|'email'|'code'|'default',
   *    label: string,
   *    placeholder: string,
   *    hint: string | Functional Component | null,
   *    required: boolean
   *  }
   * ]
   * ```
  */
  @Prop() formFields: FormFieldTypes | string[] = SIGN_IN_COMPONENTS; 

  render() {
    return (
      <amplify-form-section headerText={this.headerText} overrideStyle={this.overrideStyle} handleSubmit={this.handleSubmit}>
        <amplify-auth-fields formFields={this.formFields} />
        <div slot="amplify-form-section-footer">
          <SignInFormFooter submitButtonText={this.submitButtonText} createAccountText={CREATE_ACCOUNT_TEXT} noAccountText={NO_ACCOUNT_TEXT} />
        </div>
      </amplify-form-section>
    );
  }
}
