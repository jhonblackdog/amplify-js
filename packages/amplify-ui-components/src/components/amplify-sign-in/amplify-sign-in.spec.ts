import { newSpecPage } from '@stencil/core/testing';
import { AmplifySignIn } from './amplify-sign-in';
import { AmplifyForgotPasswordHint } from './amplify-forgot-password-hint';

describe('amplify-sign-in spec:', () => {
  describe('Component logic ->', () => {
    let signIn;

    beforeEach(() => {
      signIn = new AmplifySignIn();
    });

    it('should render `SIGN IN COMPONENTS` on form fields by default', () => {

      const result = [
        {
          type: 'username',
          required: true,
        },
        {
          type: 'password',
          hint: {
            "$attrs$": {
              "forgotPasswordText": "Forgot your password?",
              "resetPasswordText": "Reset password",
            },
            "$children$": null,
            "$elm$": undefined,
            "$flags$": 0,
            "$tag$": AmplifyForgotPasswordHint,
          },
          required: true,
        }
      ];

      expect(signIn.formFields).toEqual(result);
    });

    it('should render `handleSubmit` as undefined by default', () => {
      expect(signIn.handleSubmit).toBeUndefined();
    });

    it('should render `validationErrors` as undefined by default', () => {
      expect(signIn.validationErrors).toBeUndefined();
    });

    it('should render `headerText` to `Sign into your account` by default', () => {
      expect(signIn.headerText).toEqual('Sign into your account');
    });

    it('should render `submitButtonText` to `Sign in` by default', () => {
      expect(signIn.submitButtonText).toEqual('Sign in');
    });

    it('should render `overrideStyle` to false by default', () => {
      expect(signIn.overrideStyle).toBe(false);
    });
  });
  describe('Render logic ->', () => {
    it('should render a `sign in` form by default', async () => {
      const page = await newSpecPage({
        components: [AmplifySignIn],
        html: `<amplify-sign-in></amplify-sign-in>`
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});