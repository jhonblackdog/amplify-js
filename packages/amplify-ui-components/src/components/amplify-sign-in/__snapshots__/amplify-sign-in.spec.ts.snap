// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sign-in spec: Render logic -> should render a \`sign in\` form by default 1`] = `
<amplify-sign-in>
  <amplify-form-section headertext="Sign into your account">
    <amplify-auth-fields></amplify-auth-fields>
    <div slot="amplify-form-section-footer">
      <div class="css-1pvt25k">
        <span>
          No account?
          <amplify-link>
            Create account
          </amplify-link>
        </span>
        <amplify-button type="submit">
          Sign in
        </amplify-button>
      </div>
    </div>
  </amplify-form-section>
</amplify-sign-in>
`;
